activerecord/lib/active_record/core.rb

```diff
@@ -192,7 +192,9 @@ def find_by(*args) # :nodoc:
         hash = args.first
 
         return super if !(Hash === hash) || hash.values.any? { |v|
-          v.nil? || Array === v || Hash === v || Relation === v || Base === v
+          case v
+          when NilClass, Array, Range, Hash, Relation, Base then true
```

Comment:
`v.class.in?(types)` is not a same with `when *types`.
`Post.new.class` is `Post`, not `Base`.
`Post.all.class` is `Post::ActiveRecord_Relation`, not `Relation`.